pkg/template/controller/templateinstance_finalizer.go

```diff
@@ -0,0 +1,279 @@
+package controller
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	kerrs "k8s.io/apimachinery/pkg/util/errors"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	kclientsetinternal "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/utils/clock"
+
+	"github.com/golang/glog"
+
+	buildclient "github.com/openshift/origin/pkg/build/generated/internalclientset"
+	templateapi "github.com/openshift/origin/pkg/template/apis/template"
+	"github.com/openshift/origin/pkg/template/generated/informers/internalversion/template/internalversion"
+	templateclient "github.com/openshift/origin/pkg/template/generated/internalclientset"
+	templatelister "github.com/openshift/origin/pkg/template/generated/listers/template/internalversion"
+	restutil "github.com/openshift/origin/pkg/util/rest"
+)
+
+// TemplateInstanceFinalizerController watches for new TemplateInstance objects and
+// instantiates the template contained within, using parameters read from a
+// linked Secret object.  The TemplateInstanceFinalizerController instantiates objects
+// using its own service account, first verifying that the requester also has
+// permissions to instantiate.
+type TemplateInstanceFinalizerController struct {
+	restmapper        meta.RESTMapper
+	dynamicRestMapper *discovery.DeferredDiscoveryRESTMapper
```

Comment:
why do you need to know it's a particular kind of restmapper?

Reply:
OCD :)

will change to interface.